### PR TITLE
[DEV-3891] Combine compatible tile queries for historical features

### PR DESCRIPTION
**Pull request by YS-L**
## Description

Combine compatible tile queries for historical features.

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: tests/unit/query_graph/conftest.py [hunk 1]
```diff
@@ -130,6 +130,8 @@ def input_node_fixture(global_graph, input_details):
         "columns": [
             {"name": "ts", "dtype": DBVarType.TIMESTAMP},
             {"name": "cust_id", "dtype": DBVarType.INT},
+            {"name": "biz_id", "dtype": DBVarType.INT},
+            {"name": "product_type", "dtype": DBVarType.INT},
```
Comment (YS-L) [inline tests/unit/query_graph/conftest.py:134]: Adding these columns in the input node changes the aggregation_id in the tests. Previously, tile compute queries in the tests referenced non-existent columns as groupby keys. With this PR, the columns now need to exist due to changes in how the queries are constructed.

File: tests/fixtures/query_graph/expected_combined_tile_query.sql [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  COUNT("input_col_avg_7878f6dd82c857a14e65c8c50286995e4ca267ec") AS count_value_avg_7878f6dd82c857a14e65c8c50286995e4ca267ec
+FROM (
+  SELECT
+    *,
+    F_TIMESTAMP_TO_INDEX(CONVERT_TIMEZONE('UTC', "event_timestamp"), 300, 600, 30) AS index
+  FROM __FB_TILE_COMPUTE_INPUT_TABLE_NAME
+)
+GROUP BY
+  index,
+  "cust_id"
```
Comment (YS-L) [inline tests/fixtures/query_graph/expected_combined_tile_query.sql:33]: This is an example of a combined tile query.

File: featurebyte/query_graph/sql/interpreter/tile.py [hunk 3]
```diff
@@ -74,6 +75,17 @@ class TileGenSql:
     parent: str | None
     agg_func: str | None
 
+    @property
```
Comment (kchua78) [inline featurebyte/query_graph/sql/interpreter/tile.py:78]: Do we need this property? It does not seem to be used

Comment (YS-L) [reply featurebyte/query_graph/sql/interpreter/tile.py:78]: It's used in a few places: 

image

**Review by gee-senbong — APPROVED** [review]

**Pull request merged**